core/src/stores/postgres.rs

```diff
@@ -92,49 +92,58 @@ impl PostgresStore {
 
     fn where_clauses_and_params_for_filter<'a>(
         filter: &'a Option<SearchFilter>,
+        tags_column: Option<&str>,
+        parents_column: Option<&str>,
+        timestamp_column: Option<&str>,
         from_idx: usize,
     ) -> (Vec<String>, Vec<&'a (dyn ToSql + Sync)>, usize) {
         let mut where_clauses: Vec<String> = vec![];
         let mut params: Vec<&'a (dyn ToSql + Sync)> = vec![];
         let mut p_idx: usize = from_idx;
 
         if let Some(filter) = filter {
-            if let Some(tags_filter) = &filter.tags {
-                if let Some(tags) = &tags_filter.is_in {
-                    where_clauses.push(format!("tags_array && ${}", p_idx));
-                    params.push(tags as &(dyn ToSql + Sync));
-                    p_idx += 1;
-                }
-                if let Some(tags) = &tags_filter.is_not {
-                    where_clauses.push(format!("NOT tags_array && ${}", p_idx));
-                    params.push(tags as &(dyn ToSql + Sync));
-                    p_idx += 1;
+            if let Some(tags_column) = tags_column {
+                if let Some(tags_filter) = &filter.tags {
+                    if let Some(tags) = &tags_filter.is_in {
+                        where_clauses.push(format!("{} && ${}", tags_column, p_idx));
+                        params.push(tags as &(dyn ToSql + Sync));
+                        p_idx += 1;
+                    }
+                    if let Some(tags) = &tags_filter.is_not {
+                        where_clauses.push(format!("NOT {} && ${}", tags_column, p_idx));
+                        params.push(tags as &(dyn ToSql + Sync));
+                        p_idx += 1;
+                    }
                 }
             }
 
-            if let Some(parents_filter) = &filter.parents {
-                if let Some(parents) = &parents_filter.is_in {
-                    where_clauses.push(format!("parents && ${}", p_idx));
-                    params.push(parents as &(dyn ToSql + Sync));
-                    p_idx += 1;
-                }
-                if let Some(parents) = &parents_filter.is_not {
-                    where_clauses.push(format!("NOT parents && ${}", p_idx));
-                    params.push(parents as &(dyn ToSql + Sync));
-                    p_idx += 1;
+            if let Some(parents_column) = parents_column {
```

Comment:
This is dangerous not passing parents would deactiate parent filtering which would potentially have drastic security consequence. Any chance we can make parents_column non optional?

Reply:
Looks like we can make parents and timestamp non optional right?

Reply:
Yes parents and timestamp are always supposed to be set, so i can get rid of the optional